# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libGLESv2/libGLESv2_autogen.def
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libGLESv2/libGLESv2_autogen.def
--- a/gfx/angle/checkout/src/libGLESv2/libGLESv2_autogen.def	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libGLESv2/libGLESv2_autogen.def	Sat Sep 07 13:38:36 2019 +0000
@@ -508,6 +508,10 @@
     glTexSubImage2DRobustANGLE
     glTexSubImage3DRobustANGLE
 
+    ; GL_ANGLE_texture_external_update
+    glInvalidateTextureANGLE
+    glTexImage2DExternalANGLE
+
     ; GL_ANGLE_texture_multisample
     glGetMultisamplefvANGLE
     glGetTexLevelParameterfvANGLE
@@ -533,6 +537,9 @@
     glMatrixLoadIdentityCHROMIUM
     glMatrixLoadfCHROMIUM
 
+    ; GL_CHROMIUM_lose_context
+    glLoseContextCHROMIUM
+
     ; GL_CHROMIUM_path_rendering
     glBindFragmentInputLocationCHROMIUM
     glCoverFillPathCHROMIUM
@@ -600,6 +607,23 @@
     glFlushMappedBufferRangeEXT
     glMapBufferRangeEXT
 
+    ; GL_EXT_memory_object
+    glBufferStorageMemEXT
+    glCreateMemoryObjectsEXT
+    glDeleteMemoryObjectsEXT
+    glGetMemoryObjectParameterivEXT
+    glGetUnsignedBytei_vEXT
+    glGetUnsignedBytevEXT
+    glIsMemoryObjectEXT
+    glMemoryObjectParameterivEXT
+    glTexStorageMem2DEXT
+    glTexStorageMem2DMultisampleEXT
+    glTexStorageMem3DEXT
+    glTexStorageMem3DMultisampleEXT
+
+    ; GL_EXT_memory_object_fd
+    glImportMemoryFdEXT
+
     ; GL_EXT_occlusion_query_boolean
 
     ; GL_EXT_robustness
@@ -608,6 +632,18 @@
     glGetnUniformivEXT
     glReadnPixelsEXT
 
+    ; GL_EXT_semaphore
+    glDeleteSemaphoresEXT
+    glGenSemaphoresEXT
+    glGetSemaphoreParameterui64vEXT
+    glIsSemaphoreEXT
+    glSemaphoreParameterui64vEXT
+    glSignalSemaphoreEXT
+    glWaitSemaphoreEXT
+
+    ; GL_EXT_semaphore_fd
+    glImportSemaphoreFdEXT
+
     ; GL_EXT_texture_storage
     glTexStorage1DEXT
     glTexStorage2DEXT
@@ -690,6 +726,14 @@
     ; GL_OES_query_matrix
     glQueryMatrixxOES
 
+    ; GL_OES_texture_3D
+    glCompressedTexImage3DOES
+    glCompressedTexSubImage3DOES
+    glCopyTexSubImage3DOES
+    glFramebufferTexture3DOES
+    glTexImage3DOES
+    glTexSubImage3DOES
+
     ; GL_OES_texture_border_clamp
     glGetSamplerParameterIivOES
     glGetSamplerParameterIuivOES
@@ -762,6 +806,7 @@
     glBlitFramebufferContextANGLE
     glBlitFramebufferANGLEContextANGLE
     glBufferDataContextANGLE
+    glBufferStorageMemEXTContextANGLE
     glBufferSubDataContextANGLE
     glCheckFramebufferStatusContextANGLE
     glCheckFramebufferStatusOESContextANGLE
@@ -789,10 +834,12 @@
     glCompressedTexImage2DContextANGLE
     glCompressedTexImage2DRobustANGLEContextANGLE
     glCompressedTexImage3DContextANGLE
+    glCompressedTexImage3DOESContextANGLE
     glCompressedTexImage3DRobustANGLEContextANGLE
     glCompressedTexSubImage2DContextANGLE
     glCompressedTexSubImage2DRobustANGLEContextANGLE
     glCompressedTexSubImage3DContextANGLE
+    glCompressedTexSubImage3DOESContextANGLE
     glCompressedTexSubImage3DRobustANGLEContextANGLE
     glCopyBufferSubDataContextANGLE
     glCopySubTexture3DANGLEContextANGLE
@@ -800,6 +847,7 @@
     glCopyTexImage2DContextANGLE
     glCopyTexSubImage2DContextANGLE
     glCopyTexSubImage3DContextANGLE
+    glCopyTexSubImage3DOESContextANGLE
     glCopyTexture3DANGLEContextANGLE
     glCopyTextureCHROMIUMContextANGLE
     glCoverFillPathCHROMIUMContextANGLE
@@ -807,6 +855,7 @@
     glCoverStrokePathCHROMIUMContextANGLE
     glCoverStrokePathInstancedCHROMIUMContextANGLE
     glCoverageModulationCHROMIUMContextANGLE
+    glCreateMemoryObjectsEXTContextANGLE
     glCreateProgramContextANGLE
     glCreateShaderContextANGLE
     glCreateShaderProgramvContextANGLE
@@ -819,6 +868,7 @@
     glDeleteFencesNVContextANGLE
     glDeleteFramebuffersContextANGLE
     glDeleteFramebuffersOESContextANGLE
+    glDeleteMemoryObjectsEXTContextANGLE
     glDeletePathsCHROMIUMContextANGLE
     glDeleteProgramContextANGLE
     glDeleteProgramPipelinesContextANGLE
@@ -827,6 +877,7 @@
     glDeleteRenderbuffersContextANGLE
     glDeleteRenderbuffersOESContextANGLE
     glDeleteSamplersContextANGLE
+    glDeleteSemaphoresEXTContextANGLE
     glDeleteShaderContextANGLE
     glDeleteSyncContextANGLE
     glDeleteTexturesContextANGLE
@@ -888,6 +939,7 @@
     glFramebufferRenderbufferOESContextANGLE
     glFramebufferTexture2DContextANGLE
     glFramebufferTexture2DOESContextANGLE
+    glFramebufferTexture3DOESContextANGLE
     glFramebufferTextureEXTContextANGLE
     glFramebufferTextureLayerContextANGLE
     glFramebufferTextureMultiviewOVRContextANGLE
@@ -905,6 +957,7 @@
     glGenRenderbuffersContextANGLE
     glGenRenderbuffersOESContextANGLE
     glGenSamplersContextANGLE
+    glGenSemaphoresEXTContextANGLE
     glGenTexturesContextANGLE
     glGenTransformFeedbacksContextANGLE
     glGenVertexArraysContextANGLE
@@ -960,6 +1013,7 @@
     glGetLightxvContextANGLE
     glGetMaterialfvContextANGLE
     glGetMaterialxvContextANGLE
+    glGetMemoryObjectParameterivEXTContextANGLE
     glGetMultisamplefvContextANGLE
     glGetMultisamplefvANGLEContextANGLE
     glGetMultisamplefvRobustANGLEContextANGLE
@@ -1007,6 +1061,7 @@
     glGetSamplerParameterfvRobustANGLEContextANGLE
     glGetSamplerParameterivContextANGLE
     glGetSamplerParameterivRobustANGLEContextANGLE
+    glGetSemaphoreParameterui64vEXTContextANGLE
     glGetShaderInfoLogContextANGLE
     glGetShaderPrecisionFormatContextANGLE
     glGetShaderSourceContextANGLE
@@ -1047,6 +1102,8 @@
     glGetUniformivRobustANGLEContextANGLE
     glGetUniformuivContextANGLE
     glGetUniformuivRobustANGLEContextANGLE
+    glGetUnsignedBytei_vEXTContextANGLE
+    glGetUnsignedBytevEXTContextANGLE
     glGetVertexAttribIivContextANGLE
     glGetVertexAttribIivRobustANGLEContextANGLE
     glGetVertexAttribIuivContextANGLE
@@ -1063,14 +1120,18 @@
     glGetnUniformivRobustANGLEContextANGLE
     glGetnUniformuivRobustANGLEContextANGLE
     glHintContextANGLE
+    glImportMemoryFdEXTContextANGLE
+    glImportSemaphoreFdEXTContextANGLE
     glInsertEventMarkerEXTContextANGLE
     glInvalidateFramebufferContextANGLE
     glInvalidateSubFramebufferContextANGLE
+    glInvalidateTextureANGLEContextANGLE
     glIsBufferContextANGLE
     glIsEnabledContextANGLE
     glIsFenceNVContextANGLE
     glIsFramebufferContextANGLE
     glIsFramebufferOESContextANGLE
+    glIsMemoryObjectEXTContextANGLE
     glIsPathCHROMIUMContextANGLE
     glIsProgramContextANGLE
     glIsProgramPipelineContextANGLE
@@ -1079,6 +1140,7 @@
     glIsRenderbufferContextANGLE
     glIsRenderbufferOESContextANGLE
     glIsSamplerContextANGLE
+    glIsSemaphoreEXTContextANGLE
     glIsShaderContextANGLE
     glIsSyncContextANGLE
     glIsTextureContextANGLE
@@ -1101,6 +1163,7 @@
     glLoadMatrixxContextANGLE
     glLoadPaletteFromModelViewMatrixOESContextANGLE
     glLogicOpContextANGLE
+    glLoseContextCHROMIUMContextANGLE
     glMapBufferOESContextANGLE
     glMapBufferRangeContextANGLE
     glMapBufferRangeEXTContextANGLE
@@ -1115,6 +1178,7 @@
     glMaxShaderCompilerThreadsKHRContextANGLE
     glMemoryBarrierContextANGLE
     glMemoryBarrierByRegionContextANGLE
+    glMemoryObjectParameterivEXTContextANGLE
     glMultMatrixfContextANGLE
     glMultMatrixxContextANGLE
     glMultiDrawArraysANGLEContextANGLE
@@ -1222,10 +1286,12 @@
     glScalefContextANGLE
     glScalexContextANGLE
     glScissorContextANGLE
+    glSemaphoreParameterui64vEXTContextANGLE
     glSetFenceNVContextANGLE
     glShadeModelContextANGLE
     glShaderBinaryContextANGLE
     glShaderSourceContextANGLE
+    glSignalSemaphoreEXTContextANGLE
     glStencilFillPathCHROMIUMContextANGLE
     glStencilFillPathInstancedCHROMIUMContextANGLE
     glStencilFuncContextANGLE
@@ -1255,8 +1321,10 @@
     glTexGenxOESContextANGLE
     glTexGenxvOESContextANGLE
     glTexImage2DContextANGLE
+    glTexImage2DExternalANGLEContextANGLE
     glTexImage2DRobustANGLEContextANGLE
     glTexImage3DContextANGLE
+    glTexImage3DOESContextANGLE
     glTexImage3DRobustANGLEContextANGLE
     glTexParameterIivOESContextANGLE
     glTexParameterIivRobustANGLEContextANGLE
@@ -1278,9 +1346,14 @@
     glTexStorage3DContextANGLE
     glTexStorage3DEXTContextANGLE
     glTexStorage3DMultisampleOESContextANGLE
+    glTexStorageMem2DEXTContextANGLE
+    glTexStorageMem2DMultisampleEXTContextANGLE
+    glTexStorageMem3DEXTContextANGLE
+    glTexStorageMem3DMultisampleEXTContextANGLE
     glTexSubImage2DContextANGLE
     glTexSubImage2DRobustANGLEContextANGLE
     glTexSubImage3DContextANGLE
+    glTexSubImage3DOESContextANGLE
     glTexSubImage3DRobustANGLEContextANGLE
     glTransformFeedbackVaryingsContextANGLE
     glTranslatefContextANGLE
@@ -1348,6 +1421,7 @@
     glVertexBindingDivisorContextANGLE
     glVertexPointerContextANGLE
     glViewportContextANGLE
+    glWaitSemaphoreEXTContextANGLE
     glWaitSyncContextANGLE
     glWeightPointerOESContextANGLE
 
@@ -1415,6 +1489,12 @@
     EGL_GetFrameTimestampsANDROID
     EGL_GetNextFrameIdANDROID
 
+    ; EGL_ANDROID_get_native_client_buffer
+    EGL_GetNativeClientBufferANDROID
+
+    ; EGL_ANDROID_native_fence_sync
+    EGL_DupNativeFenceFDANDROID
+
     ; EGL_ANDROID_presentation_time
     EGL_PresentationTimeANDROID
 
@@ -1422,6 +1502,10 @@
     EGL_CreateDeviceANGLE
     EGL_ReleaseDeviceANGLE
 
+    ; EGL_ANGLE_feature_control
+    EGL_QueryDisplayAttribANGLE
+    EGL_QueryStringiANGLE
+
     ; EGL_ANGLE_program_cache_control
     EGL_ProgramCacheGetAttribANGLE
     EGL_ProgramCachePopulateANGLE